dace/transformation/dataflow/hbm_bank_split.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
+from typing import Any, Dict, Iterable, List, Tuple, Union
+
+from dace import data, dtypes, properties, registry
+from dace.sdfg import utils
+from dace.transformation import transformation
+from dace.sdfg import nodes as nd
+from dace import SDFG, SDFGState, memlet
+import functools
+
+
+@registry.autoregister_params(singlestate=True)
+@properties.make_properties
+class HbmBankSplit(transformation.Transformation):
+    """
+    A transformation that allows to split an array and distribute on
+    an array with one dimension more, or the reverse operation. Works in principle
+    with arbitrary arrays, but it's real use case is to distribute data on many HBM-banks.
+    Matches any 2 AccessNodes connected by any edge, if the dimensionality of the two accessed
+    arrays differ by exactly one. The sizes of the arrays have to be large enough with
+    respect to the split executed, but this is not verified. While it is allowed to use symbolics 
+    for the shapes of the array, it is expected that each dimension is divisable by the number
+    of splits specified.
+
+    Examples:
```

Comment:
You don't describe what splitting the array actually means: What happens in the SDFG when you apply this transformation?